packages/vite-plugin-svelte/src/utils/compile.js

```diff
@@ -7,7 +7,8 @@ import { log } from './log.js';
 import { createInjectScopeEverythingRulePreprocessorGroup } from './preprocess.js';
 import { mapToRelative } from './sourcemaps.js';
 
-const scriptLangRE = /<script [^>]*lang=["']?([^"' >]+)["']?[^>]*>/;
+const scriptLangRE =
+	/<script (?:[^>]*|(?:[^=>'"/]+=(?:"[^"]*"|'[^']*'|[^>\s])\s+)*)lang=["']?([^"' >]+)["']?[^>]*>/;
```

Comment:
```suggestion
	/<script (?:[^>]*|(?:[^=>'"/]+=(?:"[^"]*"|'[^']*'|[^>\s]+)\s+)*)lang=["']?([^"' >]+)["']?[^>]*>/;
```

@dummdidumm following the fixes in svelte, this has to be updated too i think

Reply:
Now that I think about it. The chances of someone doing `<script generic="T extends Record<string, any>" lang="ts"` is very low that I won't mind keeping it simple as before.

But I also don't mind the more comprehensive check. Just noting it here.